app/grandchallenge/archives/permissions.py

```diff
@@ -0,0 +1,27 @@
+from django.http import Http404
+from rest_framework import exceptions, permissions
+
+
+class ArchiveItemPermission(permissions.BasePermission):
+    """
+    Checks whether current user has permission to view or update the archive
+    corresponding to the current archive item.

```

Comment:
Do you need a separate class for this? Doesn't DRF have a concept of `get_permission_object` like we use in the Django views? We may need instead to add the permissions directly to the archive items.

Reply:
You can overwrite the `get_object()` and pass the corresponing `archive` to the permission tests, but these will still fail, because we're not using the standard permissions (view and update) and because it would check for archive_item level permissions, not archive level permisisons (even if we override the get_object() method). Without rewriting the permission logic for archive items, this seemed like the cleanest solution. Would adding permissions directly to archive items be better?

Reply:
I think that it's best to work within the framework. How do we handle this for other nested items (eg, algorithm images)? I think we set permissions directly, right?